controllers/user.controller.js

```diff
@@ -28,69 +45,90 @@ exports.getSingleUser = async (req, res) => {
         if (!mongoose.Types.ObjectId.isValid(userId)) {
             return res.status(422).json({
                 status: `422 Error`,
-                message: 'Ensure you enter a valid USER ID'
+                message: 'Ensure you enter a valid USER ID',
             });
         }
         if (!user) {
             return res.status(statusCode.NOT_FOUND).json({
                 status: `${statusCode.NOT_FOUND} Error`,
-                message: `User with the ID: ${userId} doesn't exist or has been deleted`
+                message: `User with the ID: ${userId} doesn't exist or has been deleted`,
             });
         }
         return res.status(statusCode.OK).json({
             status: `${statusCode.OK} Success`,
             message: 'User found ',
-            data: user
+            data: user,
         });
     } catch (error) {
         console.log('Error from getting user >>>> \n ', error);
+        if (error.name === 'CastError') {
+            return res.status(422).json({
+                message: 'Enter the name of the users to search',
+            });
+        }
         return res.status(statusCode.INTERNAL_SERVER_ERROR).json({
             status: `${statusCode.INTERNAL_SERVER_ERROR} Error`,
             message: 'Something went wrong. Try again'
         });
     }
 };
 
-exports.updateProfile = async (req, res) => {
+exports.editProfile = async (req, res) => {
     try {
-        const { userId } = req.user;
-        const id = req.params.userId;
-        const { address, phoneNumber, fullName, about } = req.body;
-        const image = req.file.path;
-        if (!mongoose.Types.ObjectId.isValid(id)) {
-            return res.status(422).json({
-                status: '422 Error',
-                message: 'Ensure your ID is a valid ID'
-            });
+        const { id } = req.params, 
+            image = req.file && req.file.path, 
+            {
+                address,
+                bio,
+                firstName,
+                lastName,
+                phoneNumber,
+            } = req.body;
+        const hasProfile = await Profile.findOne({ userId: id });
+  
+        if (id !== req.user._id) {
+            return res.status(403).json({ message: 'You are not allowed to edit this profile'});
         }
-        if (userId !== id) {
-            return res.status(403).json({ 
-                status: '403 Error',
-                message: "Unathorised, you can only update your account" 
-            });
+  
+        if (!hasProfile) {
+            return res.status(404).json({ message: 'You don\'t have a profile, please create a profile' });
         }
-        const user = await User.findOneAndUpdate(
+
+        const profile = await Profile.findOneAndUpdate(
+            {
+                userId: id,
+            },
+            {
+                $set: {
+                    ...req.body,
+                    image: image,
```

Comment:
you can use es7 syntax, you don't to redeclare the key and value of the object if it's the same
```suggestion
                    image,
```